src/main/java/Problem3/Book.java

```diff
@@ -15,11 +15,22 @@ public Book(String title, String author) {
 
     // copy constructor
     public Book(Book anotherBook) {
-        // homework
+        this.title = anotherBook.title;
+        this.author = anotherBook.author;
+        this.id = anotherBook.id;
     }
 
     @Override
-    public boolean equals(Object obj) {
-        // homework
+    public boolean equals(Object other) {
+        if (other.getClass() == this.getClass()) {
```

Comment:
this logic is way too complicated. try to avoid nested condition check like this.

you can negate the first level like 
```java
if (other.getClass() != this.getClass()) {
return false;
}
```